server/election/lease.go

```diff
@@ -11,39 +11,50 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package member
+package election
 
 import (
 	"context"
 	"sync/atomic"
 	"time"
 
 	"github.com/pingcap/log"
+	"github.com/pingcap/pd/v4/pkg/etcdutil"
 	"github.com/pkg/errors"
 	"go.etcd.io/etcd/clientv3"
 	"go.uber.org/zap"
 )
 
-// LeaderLease is used for renewing leadership of PD server.
-type LeaderLease struct {
-	client       *clientv3.Client
-	lease        clientv3.Lease
-	ID           clientv3.LeaseID
-	leaseTimeout time.Duration
+const (
+	revokeLeaseTimeout = time.Second
+	requestTimeout     = etcdutil.DefaultRequestTimeout
+	slowRequestTime    = etcdutil.DefaultSlowRequestTime
+)
 
-	expireTime atomic.Value
+// Lease is used for campaigning and renewing elected leadership.
```

Comment:
I didn't get how Lease is used for `campaigning` here as it still have the same method comparing to the previous `LeaderLease`. Can you explain why adding `campaigning` here?

Reply:
`Lease` is used as the low-level mechanism for campaigning and renewing elected leadership. The way to gain and maintain leadership is to update and keep the lease alive continuously. So this package wraps some `Lease` related methods that `Leadership` will call during the campaigning process.